$OpenBSD: patch-tests_Makefile,v 1.2 2012/05/30 11:09:10 jasper Exp $

First chunk:
- Ensure the libraries needed by libregress.so can also be located and are preloaded.

Second chunk:
commit 2564791c4176c19a7022762212d0afce7956e5bb
Author: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date:   Sat May 26 11:01:00 2012 +0200

    Fix regress tests on platforms where we need to look in additional
    library directories.

    If the directory does not exist it's just silently ignored.


--- tests/Makefile.orig	Tue May 22 11:38:39 2012
+++ tests/Makefile	Sat May 26 10:54:39 2012
@@ -41,7 +41,7 @@ clean :
 	rm -f $(REGRESS) $(REGRESS_OBJS) Regress-1.0.gir Regress-1.0.typelib
 
 check : all
-	cd .. && LD_LIBRARY_PATH=tests:$$LD_LIBRARY_PATH GI_TYPELIB_PATH=tests:$$GI_TYPELIB_PATH $(LUA) tests/test.lua
+	cd .. && LD_PRELOAD=/usr/lib/libpthread.so LD_LIBRARY_PATH=tests:$$LD_LIBRARY_PATH GI_TYPELIB_PATH=tests:$$GI_TYPELIB_PATH $(LUA) tests/test.lua
 
 $(REGRESS) : regress.o
 	$(CC) $(LDFLAGS) -o $@ regress.o $(LIBS)
@@ -57,6 +57,8 @@ Regress-1.0.gir : $(REGRESS)
 	g-ir-scanner --warn-all --no-libtool --quiet --output=$@	\
 	  --namespace=Regress --nsversion=1.0				\
 	  --include=cairo-1.0 --include=Gio-2.0				\
+	  --library-path=/usr/lib --library-path=${X11BASE}/lib		\
+	  --library-path=${LOCALBASE}/lib				\
 	  $(GIDATADIR)/regress.c $(GIDATADIR)/regress.h			\
 	 -lregress
 
